server/src/main/java/org/elasticsearch/client/transport/TransportClient.java

```diff
@@ -150,8 +151,10 @@ private static ClientTemplate buildTemplate(Settings providedSettings, Settings
             SettingsModule settingsModule = new SettingsModule(settings, additionalSettings, additionalSettingsFilter);
 
             SearchModule searchModule = new SearchModule(settings, true, pluginsService.filterPlugins(SearchPlugin.class));
+            IndicesModule indicesModule = new IndicesModule(Collections.emptyList());
             List<NamedWriteableRegistry.Entry> entries = new ArrayList<>();
             entries.addAll(NetworkModule.getNamedWriteables());
+            entries.addAll(indicesModule.getNamedWriteables());
```

Comment:
A small nit here; we have the ordering `searchModule, indicesModule` for the local variables; can we do the same on the invocation to `getNamedWritables`?